### DISCUSSION TITLE
Grayscale via querystring parameter

### DISCUSSION THREAD

**Discussion by danbramall** [SixLabors/ImageSharp.Web#310] [Q&A] [answered]
Hi,

I'm wondering if it's possible to have an image returned in grayscale via a querystring parameter? e.g. Something like `image.png?width=100&grayscale=true&format=webp`.

I can see there's a [GrayscaleExtensions class](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.GrayscaleExtensions.html) in the core library but presumably I'd need to build my own middleware if it's not possible to use that 'out of the box' with ImageSharp.Web.

I'm a front-end dev currently working on an Umbraco project so was hoping there'd be something like the old ImageProcessor filters (https://imageprocessor.org/imageprocessor/imagefactory/filter/).

Many thanks for any pointers :)

**Comment by JimBobSquarePants** [comment, marked as answer]
Hi @danbramall 

@stefannikolei opened a PR to add this but I'm not going to add it to the core processor list so I can keep the potential attack surface at a minimum for DDOS attacks. 

You can use this as a basis for an implementation though. The library is designed to be extensible to you can bring your own processing commands.

https://github.com/SixLabors/ImageSharp.Web/pull/313

**Reply by danbramall** [reply]
Cheers @JimBobSquarePants, sounds sensible to me. I'll have a look at a custom implementation.
